src/rpdk/core/project.py

```diff
@@ -217,15 +223,37 @@ def generate(self):
         # generate template for IAM role assumed by cloudformation
         # to provision resources if schema has handlers defined
         if "handlers" in self.schema:
+            handlers = self.schema["handlers"]
             template = self.env.get_template("resource-role.yml")
             permission = "Allow"
             path = self.root / ROLE_TEMPLATE_FILENAME
             LOG.debug("Writing Resource Role CloudFormation template: %s", path)
             actions = {
                 action
-                for handler in self.schema["handlers"].values()
+                for handler in handlers.values()
                 for action in handler.get("permissions", [])
             }
+
+            # calculate IAM role max session timeout based on highest handler timeout
+            # with some buffer (70 seconds per minute)
+            cud_handlers = [
+                handlers.get(handler, {})
+                for handler in ["create", "update", "delete"]
+                if handlers.get(handler, None)
+            ]
+            max_handler_timeout = max(
+                (
+                    handler.get("timeoutInMinutes", DEFAULT_ROLE_TIMEOUT_MINUTES)
+                    for handler in cud_handlers
```

Comment:
if you just loop over all handler values you can avoid the filtering above with the cud_handlers. The read and list would just default to the default time since it wont be found.

Reply:
we don't want it to be the default time if all values specified are less than the default time